### PR TITLE
Refactor ride measurement storage

### PR DESCRIPTION
**Pull request by IntelOrca**
* Maximum number of simultaneous ride measurements can be increased at will.
* List of ride measurements removed, now just a pointer in `struct Ride`.
* Ride measurements now dynamically allocated on heap when required (~19 KiB per ride).

### CODE REVIEW

File: src/openrct2/rct2/S6Importer.cpp [hunk 1]
```diff
@@ -750,6 +748,37 @@ class S6Importer final : public IParkImporter
         // pad_208[0x58];
     }
 
+    void ImportRideMeasurements()
+    {
+        for (const auto& src : _s6.ride_measurements)
+        {
+            if (src.ride_index != 255)
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp]:
```suggestion
            if (src.ride_index != RIDE_ID_NULL)
```

File: src/openrct2/rct1/S4Importer.cpp [hunk 2]
```diff
@@ -1108,23 +1107,32 @@ class S4Importer final : public IParkImporter
 
     void ImportRideMeasurements()
     {
-        for (int32_t i = 0; i < MAX_RIDE_MEASUREMENTS; i++)
+        for (const auto& src : _s4.ride_measurements)
         {
-            rct_ride_measurement* dst = get_ride_measurement(i);
-            rct_ride_measurement* src = &_s4.ride_measurements[i];
-            ImportRideMeasurement(dst, src);
+            if (src.ride_index != 255)
```
Comment (ZehMatt) [inline src/openrct2/rct1/S4Importer.cpp]:
```suggestion
            if (src.ride_index != RIDE_ID_NULL)
```

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -3134,136 +3100,106 @@ void ride_measurements_update()
...
+        if (numRideMeasurements > MAX_RIDE_MEASUREMENTS && lruRide != nullptr)
         {
-            measurement = get_ride_measurement(i);
+            lruRide->measurement = {};
+            numRideMeasurements--;
         }
+    } while (numRideMeasurements > MAX_RIDE_MEASUREMENTS);
+}
+
+std::pair<RideMeasurement*, rct_string_id> ride_get_measurement(Ride* ride)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp]: Shouldn't this become a ride member function.

File: src/openrct2/actions/RideDemolishAction.hpp [hunk 4]
```diff
@@ -229,10 +229,6 @@ DEFINE_GAME_ACTION(RideDemolishAction, GAME_COMMAND_DEMOLISH_RIDE, GameActionRes
             }
         }
 
-        user_string_free(ride->name);
-        ride->type = RIDE_TYPE_NULL;
```
[the remark this answers is not in the document]
Comment (IntelOrca) [reply src/openrct2/actions/RideDemolishAction.hpp]: Yes

File: src/openrct2/ride/Ride.cpp (hunk 3 above)
Comment (IntelOrca) [reply src/openrct2/ride/Ride.cpp]: I couldn't decide at the time whether it should be or not.

**Pull request merged**